xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAEnumeratorHD.cpp

```diff
@@ -440,3 +420,314 @@ bool CEnumeratorHD::IsSDRSupported()
 
   return true;
 }
+
+bool CEnumeratorHD::GetProcessorFormats(std::vector<DXGI_FORMAT>* inputFormats,
+                                        std::vector<DXGI_FORMAT>* outputFormats) const
+{
+  // Not initialized yet
+  if (!m_pEnumerator)
+    return false;
+
+  std::vector<DXGI_FORMAT> result;
+
+  HRESULT hr;
+  UINT uiFlags;
+  for (int fmt = DXGI_FORMAT_UNKNOWN; fmt <= DXGI_FORMAT_V408; fmt++)
+  {
+    DXGI_FORMAT dxgiFormat = static_cast<DXGI_FORMAT>(fmt);
+    if (S_OK == (hr = m_pEnumerator->CheckVideoProcessorFormat(dxgiFormat, &uiFlags)))
+    {
+      if ((uiFlags & D3D11_VIDEO_PROCESSOR_FORMAT_SUPPORT_INPUT) && inputFormats)
+        inputFormats->push_back(dxgiFormat);
+      if ((uiFlags & D3D11_VIDEO_PROCESSOR_FORMAT_SUPPORT_OUTPUT) && outputFormats)
+        outputFormats->push_back(dxgiFormat);
+    }
+    else
+    {
+      CLog::LogF(LOGWARNING,
+                 "Unable to retrieve support of the dxva processor for format {}, error {}",
+                 DX::DXGIFormatToString(dxgiFormat), DX::GetErrorDescription(hr));
+      return false;
+    }
+  }
+  return true;
+}
+
+std::vector<DXGI_FORMAT> CEnumeratorHD::GetProcessorRGBOutputFormats()
+{
+  std::vector<DXGI_FORMAT> outputFormats;
```

Comment:
maybe use reserve to avoid several memory reallocations (if is know proximate maximum)

Reply:
Good idea, It could pre-allocate the same number of elements as the original vector. On AMD almost all formats are RGB-like. On Intel it's half, or less with the most recent gpu.

Another approach would be to provide a filter predicate to GetProcessorFormats as optional parameter to filter at the source. Worth it? It's not like high performance is required from this function.